### PR TITLE
.Net - Fix agent function invocation and examples

### PR DESCRIPTION
**Pull request by crickman**
### Motivation and Context

Noticed that agent examples failing due to resourcing and configuration delta.  As part of this discovered agent as plugin regression from community contribution.

### Description

- Include agent templates as embedded resources
- Switch from `TestConfiguration.AzureOpenAI.DeploymentName` to `TestConfiguration.AzureOpenAI.ChatDeploymentName` in `Example72_AgentCollaboration`
- Remove `fileIds` from `Agent.AskAsync` (not utilized and resulting in failed run creation) - https://github.com/microsoft/semantic-kernel/pull/5110
- Eliminated user-message duplication
- Added comments to `ChatMessageType`

image

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/samples/KernelSyntaxExamples/KernelSyntaxExamples.csproj [hunk 1]
```diff
@@ -83,13 +75,25 @@
...
     <None Include="Plugins\ApiManifestPlugins\**\apimanifest.json">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
   </ItemGroup>
+  <ItemGroup>
+    <EmbeddedResource Include="Resources\*">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </EmbeddedResource>
+    <EmbeddedResource Include="Resources\Agents\ParrotAgent.yaml" />
+    <EmbeddedResource Include="Resources\Agents\ToolAgent.yaml" />
```
Comment (markwallace-microsoft) [inline dotnet/samples/KernelSyntaxExamples/KernelSyntaxExamples.csproj:97]: Should we use wildcards or can the other statement we extended to include subdirectories?

**Review by markwallace-microsoft — APPROVED** [review]

File: dotnet/samples/KernelSyntaxExamples/KernelSyntaxExamples.csproj (hunk 1 above)
Comment (crickman) [reply dotnet/samples/KernelSyntaxExamples/KernelSyntaxExamples.csproj:97]: I wondered on whether to use a greedy expression for sub-directories.  I think I'd prefer `Resources\Agents\*` for control.  

I guess when I saw the project was already asymetrical (with the `None` entries), it chose not to overthink.

Comment (crickman) [reply dotnet/samples/KernelSyntaxExamples/KernelSyntaxExamples.csproj]: Just noticed, looks like `**` is greedy:

image

Comment (crickman) [reply dotnet/samples/KernelSyntaxExamples/KernelSyntaxExamples.csproj]: Ok, I don't think the "None" entries are doing anything...removed.

**Review by TaoChenOSU — APPROVED** [review]

**Pull request merged**